client/src/components/transformation/merge/SourceMergeOptions.jsx

```diff
@@ -99,24 +105,26 @@ function SelectMergeColumns({ onChange, onChangeAll, columns, selected }) {
           {selected.size === columns.size ? 'All columns selected' : 'Select all Columns'}
         </fieldset>
       </legend>
-      {columns.map((column) => {
-        const cName = columnName(column);
-
-        const id = `merge_column_${cName}`;
-        return (
-          <div key={cName}>
-            <input
-              type="checkbox"
-              id={id}
-              name="merge_column"
-              value={cName}
-              selected={selectedNames.includes(cName)}
-              checked={selectedNames.includes(cName)}
-              onChange={() => onChange(column)}
-            />
-            <label htmlFor={id}>{column.get('title')}</label>
-          </div>
-        );
+      {orderedGroups.map((groupName) => {
+        const colsData = columns.filter(column => column.get('groupName') === groupName).map((column) => {
+          const cName = columnName(column);
+          const id = `merge_column_${cName}`;
+          return (
+            <div key={cName}>
+              <input
+                type="checkbox"
+                id={id}
+                name="merge_column"
+                value={cName}
+                selected={selectedNames.includes(cName)}
+                checked={selectedNames.includes(cName)}
+                onChange={() => onChange(column)}
+              />
+              <label htmlFor={id}>{column.get('title')}</label>
+            </div>
+          );
+        });
+        return <div key={groupName}><div className="groupName">{capitalizeFirstLetter(groupName)}</div>{colsData}</div>;
```

Comment:
I would suggest to add a different *style*, similar of what we do with drop-downs. There the group names are *smaller*
![2020-12-12-204418_1920x1080_scrot](https://user-images.githubusercontent.com/178474/101993440-1394c580-3cbb-11eb-8977-06ee832b5795.png)

Reply:
Not sure now if *smaller* or **bold** is the way to go.